layers/cmd/valueset.go

```diff
@@ -311,6 +319,150 @@ func LoadValuesetFiles(vs *Valuesets, files []string) error {
 	return nil
 }
 
+type Options struct {
+	lookupOrder []string
+	output      []string
+	separator   map[string]string
+}
+
+func (opt Options) splitCell(header, cell string) []string {
+	sep := opt.separator[header]
+	// if separator is empty, default to space
+	if sep == "" || sep == " " {
+		return strings.Split(strings.TrimSpace(cell), " ")
+	}
+	s := strings.ReplaceAll(cell, " ", "")
+	return strings.Split(strings.TrimSpace(s), sep)
+}
+
+var options_re = regexp.MustCompile(`options`)
+
+func parseSpreadsheet(sheet [][]string) (optionsRows [][]string, headers []string, data [][]string, err error) {
+	var headerIdx int
+ROWS:
+	for rowIdx, row := range sheet {
+		if len(row) == 0 {
+			continue
+		}
+		for _, columnData := range row {
+			if options_re.MatchString(columnData) {
+				optionsRows = append(optionsRows, row)
+				continue ROWS
+			} else {
+				headers = row
+				headerIdx = rowIdx
+				break ROWS
+			}
+		}
+	}
+	data = sheet[headerIdx+1:][:]
+	if len(data) == 0 {
+		err = fmt.Errorf("Empty data")
+	}
+	return optionsRows, headers, data, err
+}
+
+func parseOptions(optionsRows [][]string) Options {
+	options := Options{
+		lookupOrder: make([]string, 0),
+		output:      make([]string, 0),
+		separator:   map[string]string{},
+	}
+	for _, opt := range optionsRows {
+		optionType := opt[0]
+		switch optionType {
+		case "options.lookupOrder":
+			options.lookupOrder = opt[1:]
+		case "options.output":
+			options.output = opt[1:]
+		case "options.separator":
+			// options.separator | DESCRIPTIVE_TEXT | ;
+			options.separator[opt[1]] = opt[2]
+		}
+	}
+	return options
+}
+
+func parseData(sheetName string, headers []string, data [][]string, options Options) (Valuesets, error) {
+	vs := Valuesets{Sets: map[string]Valueset{}}
+	vs.Sets[sheetName] = Valueset{}
+	var entry = vs.Sets[sheetName]
+
+	for _, row := range data {
+		for colIdx, col := range row {
+			header := headers[colIdx]
+			var headerSeparatorCount int
+			// if header column has a separator, treat every cell in this as if it has a separator
+			if _, exists := options.separator[header]; exists {
```

Comment:
Header having a separator is not a special case. split should return []string{cell} if there is no separator.